cmd/lakectl/cmd/sst.go

```diff
@@ -1,70 +1,44 @@
 package cmd
 
 import (
-	"bytes"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
-	"time"
 
-	sst "github.com/cockroachdb/pebble/sstable"
+	pebblesst "github.com/cockroachdb/pebble/sstable"
 	"github.com/spf13/cobra"
 	"github.com/treeverse/lakefs/catalog/rocks"
 	"github.com/treeverse/lakefs/graveler/committed"
 	"github.com/treeverse/lakefs/graveler/sstable"
 )
 
-type InMemFileInfo struct {
-	name string
-	size int64
-}
-
-func (r *InMemFileInfo) Name() string       { return r.name }
-func (r *InMemFileInfo) Size() int64        { return r.size }
-func (r *InMemFileInfo) Mode() os.FileMode  { return os.ModePerm }
-func (r *InMemFileInfo) ModTime() time.Time { return time.Now() }
-func (r *InMemFileInfo) IsDir() bool        { return false }
-func (r *InMemFileInfo) Sys() interface{}   { return nil }
-
-type InMemReadableFile struct {
-	data *bytes.Reader
-}
-
-func (r *InMemReadableFile) ReadAt(p []byte, off int64) (n int, err error) {
-	return r.data.ReadAt(p, off)
-}
-
-func (r *InMemReadableFile) Close() error {
-	return nil
-}
-
-func (r *InMemReadableFile) Stat() (os.FileInfo, error) {
-	return &InMemFileInfo{
-		name: "stdin",
-		size: r.data.Size(),
-	}, nil
-}
-
 func getIterFromFile(filePath string) (committed.ValueIterator, map[string]string, error) {
-	var file sst.ReadableFile
+	var file pebblesst.ReadableFile
 	var err error
 	// read from stdin (file has to be seekable/stat-able so we have this weird wrapper
 	if filePath == "-" {
-		data, err := ioutil.ReadAll(os.Stdin)
+		fh, err := ioutil.TempFile("", "cat-sst-*")
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, fmt.Errorf("error creating tempfile: %w", err)
 		}
-		file = &InMemReadableFile{
-			bytes.NewReader(data),
+		_, err = io.Copy(fh, os.Stdin)
```

Comment:
Cool.  Can you additionally add a TODO or open as issue to just keep `os.Stdin` if it is seekable?  (I think it will be enough to seek to 0 from the current position to determine whether an open file descriptor is seekable.)  It's something to make the tool a bit more convenient, but certainly not blocking.

Reply:
I assumed most people using stdin will read from a pipe but that's indeed not necessarily the case.
Implemeneted your suggestion.